src/js/components/ServiceOptions.js

```diff
@@ -0,0 +1,115 @@
+import classNames from 'classnames';
+import mixin from 'reactjs-mixin';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import CosmosPackagesStore from '../stores/CosmosPackagesStore';
+import Framework from '../structs/Framework';
+import InternalStorageMixin from '../mixins/InternalStorageMixin';
+import Service from '../structs/Service';
+
+const METHODS_TO_BIND = [
+  'handleEditConfigClick',
+  'handleConfigModalClose'
+];
+
+class ServiceOptions extends mixin(InternalStorageMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      editConfigModalOpen: false
+    };
+
+    this.internalStorage_set({
+      servicePackage: null
```

Comment:
can we just blow this up?

in every part of the code where you need to access `servicePackage`, you can go directly to the source and do `CosmosPackagesStore.getPackageDetails()` and then use that variable

if you do this, you can remove the internalStorage mixin, + all of the methods where you manually listen to the cosmos store. (`StoreMixin` got your back because it'll forceUpdate and you'll have the package on the render)